rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java

```diff
@@ -0,0 +1,147 @@
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import io.swagger.annotations.ApiOperation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import uk.gov.hmcts.ccd.definition.store.domain.service.EntityToResponseDTOMapper;
+import uk.gov.hmcts.ccd.definition.store.repository.AccessTypeRolesRepository;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.AccessTypeRolesEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseTypeEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.JurisdictionEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesField;
+import uk.gov.hmcts.ccd.definition.store.repository.model.OrganisationProfileIds;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesJurisdictionResult;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesJurisdictionResults;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesResult;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesRoleResult;
+
+import javax.validation.Valid;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@RestController
+@Api(value = "/api")
+@RequestMapping(value = "/api")
+public class AccessTypeRolesController {
+
+    private final EntityToResponseDTOMapper entityToResponseDTOMapper;
+    private final AccessTypeRolesRepository accessTypeRolesRepository;
+
+    @Autowired
+    public AccessTypeRolesController(EntityToResponseDTOMapper entityToResponseDTOMapper,
+                                     AccessTypeRolesRepository accessTypeRolesRepository) {
+        this.entityToResponseDTOMapper = entityToResponseDTOMapper;
+        this.accessTypeRolesRepository = accessTypeRolesRepository;
+    }
+
+
+    @PostMapping(value = "/retrieve-access-types",consumes = {"application/json"}, produces = {"application/json"})
+    @ApiOperation(value = "Get all accessTypes for all caseTypes", response = AccessTypeRolesJurisdictionResult.class,
+        responseContainer = "List")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "List of valid access types request"),
+        @ApiResponse(code = 401, message = "Unauthorised request"),
+        @ApiResponse(code = 403, message = "Bad request")
+    })
+    public AccessTypeRolesJurisdictionResults retrieveAccessTypeRoles(
+        @RequestBody(required = false) @Valid OrganisationProfileIds organisationProfileIds) {
+        List<AccessTypeRolesField> accessTypeRoles;
+
+        if (organisationProfileIds != null
```

Comment:
maybe core business logic should live in a service class, e.g. `AccessTypeRolesService`?

Reply:
definitely need to move this out of the controller into a service